### DISCUSSION TITLE
Are timestamps supported, or am I encoding my Parquet files incorrectly?

### DISCUSSION THREAD

**Discussion by tmcw** [duckdb/duckdb#8116] [Q&A]
Hi folks! I've been trying to generate Parquet files with _timestamps_ readable with DuckDB in the browser. Here's an example file:

- https://dropover.cloud/389060

I've gotten to the point where I generate Parquet files that, in DuckDB's CLI client, timestamps are read as dates. However, with duckdb-wasm, I can't find a way to generate Parquet files in which DuckDB-wasm reads timestamps as dates: they're always read as numbers instead. I can find examples of Parquet files with _Date_ columns that are readable, but I can't find any example of Parquet files with timestamp columns that are readable.

The code that generates this file is here: https://github.com/tmcw/bikesharecharts_collector/blob/main/src/main.rs - in the official parquet-Rust crate, I'm using the `TimestampMillisecondType`:

```rs
let mut times = PrimitiveBuilder::<TimestampMillisecondType>::new();
```

I'm not sure what I'm doing wrong, or if I'm just trying to do something that isn't possible: I have a dataset of stuff that's updated every few minutes, so using timestamps is pretty essential, but making the pieces fit together is surprisingly tricky.

**Comment by MatinF** [comment]
We seem to have issues reading the hours/minutes/seconds/ms of our Parquet timestamps via the DuckDB ODBC driver - is this not supported? It seems like only the dates are returned.
